libraries/abstractions/pkcs11/mbedtls/iot_pkcs11_mbedtls.c

```diff
@@ -2017,40 +2022,45 @@ CK_RV prvCreatePrivateKey( CK_ATTRIBUTE_PTR pxTemplate,
 
         if( lDerKeyLength < 0 )
         {
+            PKCS11_PRINT( ( "mbedTLS pk_write_key_der failed with error %s : %s \r\n",
+                            mbedtlsHighLevelCodeOrDefault( lDerKeyLength ),
+                            mbedtlsLowLevelCodeOrDefault( lDerKeyLength ) ) );
             xResult = CKR_ATTRIBUTE_VALUE_INVALID;
         }
 
         /* Clean up the mbedTLS key context. */
         mbedtls_pk_free( &xMbedContext );
     }
 
-    if( xResult == CKR_OK )
-    {
-        /* TODO: Remove this hack.
-         * mbedtls_pk_write_key_der appends empty public
-         * key data when saving EC private key
-         * that does not have a public key associated with it.
-         * a1 04        -> Application identifier of length 4
-         * 03 02     -> Bit string of length 2
-         *    00 00  -> "Public key"
-         * https://forums.mbed.com/t/how-do-i-write-an-ec-private-key-w-no-public-key-to-der-format/4728 */
-        if( xKeyType == CKK_EC ) /* CKK_EC = CKK_ECDSA. */
+    #if ( pkcs11configSUPPRESS_ECDSA_MECHANISM != 1 )
+        if( xResult == CKR_OK )
         {
-            /* If there was no public key in the structure, this byte
-             * array will be appended to the valid private key.
-             * It must be removed so that we can read the private
-             * key back at a later time. */
-            uint8_t emptyPubKey[ 6 ] = { 0xa1, 0x04, 0x03, 0x02, 0x00, 0x00 };
-            compare = memcmp( &pxDerKey[ MAX_LENGTH_KEY - 6 ], emptyPubKey, 6 );
-
-            if( compare == 0 )
+            /* TODO: Remove this hack.
```

Comment:
I suggest removing this comment. The work-around in place below has proven to be sufficient.

Reply:
Done, removed this comment